pcdsdevices/lodcm.py

```diff
@@ -60,6 +69,437 @@ def __init__(self, prefix, *args, **kwargs):
+    """
+    # x, y, and z are on the base but not touched in normal operations
+    z1 = Cpt(IMS, ':MON:MMS:04', kind='normal', doc='LOM Xtal1 Z')
+    x1 = Cpt(IMS, ':MON:MMS:05', kind='normal', doc='LOM Xtal1 X')
+    y1 = Cpt(IMS, ':MON:MMS:06', kind='normal', doc='LOM Xtal1 Y')
+    # theta movement
+    th1 = Cpt(IMS, ':MON:MMS:07', kind='normal', doc='LOM Xtal1 Theta')
+    # chi movement
+    ch1 = Cpt(IMS, ':MON:MMS:08', kind='normal', doc='LOM Xtal1 Chi')
+    # normal to the crystal surface movement
+    h1n = Cpt(IMS, ':MON:MMS:09', kind='normal', doc='LOM Xtal1 Hn')
+    # paralell to the crystal surface movement
+    h1p = Cpt(IMS, ':MON:MMS:20', kind='normal', doc='LOM Xtal1 Hp')
+
+    tab_component_names = True
+
+    def __init__(self, prefix, *args, **kwargs):
+        super().__init__(prefix, *args, **kwargs)
+
+
+class CrystalTower2(Device):
+    """
+    Crystal Tower 2.
+
+    Parameters
+    ----------
+    prefix : str
+        Epics base Pv prefix.
+    """
+    # x, y, and z are on the base but not touched in normal operations
+    z2 = Cpt(IMS, ':MON:MMS:10', kind='normal', doc='LOM Xtal2 Z')
+    x2 = Cpt(IMS, ':MON:MMS:11', kind='normal', doc='LOM Xtal2 X')
+    y2 = Cpt(IMS, ':MON:MMS:12', kind='normal', doc='LOM Xtal2 Y')
+    # thata movement
+    th2 = Cpt(IMS, ':MON:MMS:13', kind='normal', doc='LOM Xtal2 Theta')
+    # chi movement
+    ch2 = Cpt(IMS, ':MON:MMS:14', kind='normal', doc='LOM Xtal2 Chi')
+    # normal to the crystal surface movement
+    h2n = Cpt(IMS, ':MON:MMS:15', kind='normal', doc='LOM Xtal2 Hn')
+    # in the DAQ for scanning in python, only used for commissioning
+    diode2 = Cpt(IMS, ':MON:MMS:21', kind='normal', doc='LOM Xtal2 PIPS')
+
+    tab_component_names = True
+
+    def __init__(self, prefix, *args, **kwargs):
+        super().__init__(prefix, *args, **kwargs)
+
+
+class DiagnosticsTower(Device):
+    """
+    Diagnostic Tower Motors.
+
+    Parameters
+    ----------
+    prefix : str
+        Epics base PV prefix.
+    """
+    # Located midway between T1 and T2 in the center of rotation of the device.
+    # horizontal slits
+    dh = Cpt(IMS, ':MON:MMS:16', kind='normal', doc='LOM Dia H')
+    # vertical slits
+    dv = Cpt(IMS, ':MON:MMS:17', kind='normal', doc='LOM Dia V')
+    dr = Cpt(IMS, ':MON:MMS:19', kind='normal', doc='LOM Dia Theta')
+    # filters wheel
+    df = FCpt(IMS, '{self._prefix}:MON:MMS:{self._df_suffix}', kind='normal',
+              doc='LOM Dia Filter Wheel')
+    # pips diode
+    dd = Cpt(IMS, ':MON:MMS:18', kind='normal', doc='LOM Dia PIPS')
+    # yag screen
+    yag_zoom = Cpt(IMS, ':MON:CLZ:01', kind='normal', doc='LOM Zoom')
+
+    tab_component_names = True
+
+    def __init__(self, prefix, *args, **kwargs):
+        self._prefix = prefix
+        self._df_suffix = '27'
+        if 'XCS' in self._prefix:
+            self._df_suffix = '22'
+        super().__init__(prefix, *args, **kwargs)
+
+
+class OffsetIMS(PseudoSingleInterface):
```

Comment:
I think I'm not supposed to inherit from `PseudoSingleInterface`? I am not sure what the `OffsetIMS` should be

Reply:
The driving idea for `OffsetIMS` (bad name maybe?) was to let us separate the various unrelated psuedopositioner-like things from the main class, e.g. you shouldn't have to pass 10+ unrelated values through the same two `forward` and `inverse` calculation functions. Each different calculation can be independent, making it simpler to implement and maintain.

So `OffsetIMS` would be a motor that has an offset. That's it.

Two reasonable ways to implement this I think?
- `PseudoPositioner` with an `IMS` as the real motor and a `PseudoSingleInterface` as the offset motor (e.g. the Si and C motors), there might already be a good base class for this? Maybe not though.
- A subclass of `IMS` with an offset PV built in and move/readback adjusted appropriately (actually this way sounds messy)